presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTile.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.spi.PrestoException;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public final class BingTile
+{
+    public static final int MAX_ZOOM_LEVEL = 23;
+
+    private final int x;
+    private final int y;
+    private final int zoomLevel;
+
+    private BingTile(int x, int y, int zoomLevel)
+    {
+        checkArgument(zoomLevel <= MAX_ZOOM_LEVEL);
+        this.x = x;
+        this.y = y;
+        this.zoomLevel = zoomLevel;
```

Comment:
Are all x,y valid bing tiles for a given zoom level?
For example 7,7 might not be a valid x,y for zoom level 1.
Should we check for this?